### DISCUSSION TITLE
The option to sort items in folders alphabetically

### DISCUSSION THREAD

**Discussion by snden** [vkbo/novelWriter#1383] [Ideas]
Please, would it be possible to add an option to sort the contents of the Locations, Limeline, Objects, Entities and Custom folders alphabetically? For example, when I have dozens of objects in the Objects folder, it is difficult to search them without alphabetical order. Thank you.

**Comment by vkbo** [comment]
The entire point of the project tree is that it doesn't impose any specific order to the content. It would perhaps be possible to add a function to sort a notes folder alphabetically as a one off call, like the bookmarks browser in Google Chrome does. But a permanent sort option is difficult to implement since the underlying widget that displays the content either sorts everything or nothing. Any sort functionality also needs to be able to handle different languages. English sorting in non-English alphabets is just annoying.

Edit: If you have a lot of items in your Object folder, you may want to consider splitting them up in categories and putting them in subfolders. It should help finding things. At least it's what I do.

**Reply by snden** [reply]
Sorry, I explained it wrong. I meant sorting on demand - for example, an option in the context menu of the folder. Permanent alphabetical sorting is definitely not, that would not be appropriate when the author needs his own sorting.
I didn't realize the differences in alphabetical order in different languages, but English would be the solution, since it is not an exact alphabetical order, but for easier orientation in the items.

**Reply by vkbo** [reply]
It definitely needs to be able to adapt to locale. English sorting is not good enough as it is a fairly short alphabet and messes up characters that aren't included in it. Then there are the non-latin alphabets to consider. novelWriter has localisation. Even if it is not perfect, I don't intend to make it worse. But I'm sure there are solutions to be found.

Anyway, the on-demand implementation is a potential feature at least. Feel free to make a feature request.
